### ISSUE TITLE
对论文和代码的一些疑问

### ISSUE THREAD

**Issue by Linghui-Zeng** [FengZhao2001/CDS#2]
你好，我是医学图像分割的一名初学者，对于代码中中有一些疑问？
疑问1.
model = create_model(has_dropout=True)  # student model
ema_model = create_model(ema=True, has_dropout=True)  # teacher model
model1 = model 这一步导致model1和model指向同一个模型对象，实际上使两个"学生模型"共享完全相同的参数和内存地址，这是否违背了双学生网络的设计初衷呢？因为在https://github.com/ZHKKKe/DualStudent/blob/master/dual_student.py这里创建双学生模型时使用以下代码：
l_model = create_model(name='l', num_classes=num_classes)
r_model = create_model(name='r', num_classes=num_classes)
且虽然代码中创建了两个优化器：
optimizer = optim.SGD(model.parameters(), lr=base_lr, momentum=0.9, weight_decay=0.0001)
optimizer1 = optim.SGD(model1.parameters(), lr=base_lr, momentum=0.9, weight_decay=0.0001)
实际最终只使用了：
 optimizer.zero_grad()
loss.backward()
optimizer.step()
2.
       '''挑选学生A网络无标签数据的信息熵最小的前20%个'''
                prob = torch.softmax(outputs_A[labeled_bs:], dim=1)
                entropy = -torch.sum(prob * torch.log(prob + 1e-10), dim=1)
                alpha_t = 20 * (1 - epoch_num / max_epoch)
                high_thresh = np.percentile(
                    entropy[pseudo_label_studentA[labeled_bs:] != 255].cpu().detach().numpy().flatten(),
                    100 - alpha_t, )
                high_entropy_mask3 = (
                            entropy.ge(high_thresh).float() * (pseudo_label_studentA[labeled_bs:] != 255).bool())
                maskA = torch.mul(torch.mul(maskA1, maskA2), high_entropy_mask3)  # 三个掩码相乘
                '''学生B网络的高层次对比学习'''
                '''挑选学生B网络无标签数据中伪标签正确的掩码'''
                maskB1 = (pseudo_label_studentB[labeled_bs:] == pseudo_label_teacher)
                '''挑选学生A网络无标签数据的最大预测向量>的掩码'''
                outputs_normB = F.softmax(outputs_B[labeled_bs:], dim=1)
                [pred_joined_unlabeledB_MAX, indicesB] = torch.max(outputs_normB, dim=1)
                maskB2 = (pred_joined_unlabeledB_MAX > FA)
                '''挑选学生A网络无标签数据的信息熵最大的前20%个'''
                probB = torch.softmax(outputs_B[labeled_bs:], dim=1)
                entropyB = -torch.sum(probB * torch.log(probB + 1e-10), dim=1)
                high_threshB = np.percentile(
                    entropyB[pseudo_label_studentB[labeled_bs:] != 255].cpu().detach().numpy().flatten(),
                    100 - alpha_t, )
                high_entropy_maskB3 = (
                            entropyB.ge(high_threshB).float() * (pseudo_label_studentB[labeled_bs:] != 255).bool())
                maskB = torch.mul(torch.mul(maskB1, maskB2), high_entropy_maskB3)  # 三个掩码相乘
论文中对于可靠像素点描述为：像素的语义类别标签需与教师网络伪标签一致、预测置信度超过预设阈值 ϕ、信息熵低于动态阈值 ηt
但是代码中写到 high_entropy_mask3 = (entropy.ge(high_thresh).float() * (pseudo_label_studentA[labeled_bs:] != 255).bool())这一句是否没有满足第三点取信息熵更低的体素呢？且注释中写：'''挑选学生A网络无标签数据的信息熵最小的前20%个'''和'''挑选学生A网络无标签数据的信息熵最大的前20%个'''，而是实际上都选取的是信息熵高于阈值的体素，且按照您的代码逻辑，到训练后期alpha_t→0，就会导致high_entropy_mask3 和high_entropy_maskB3 全都为False了。而且信息熵越大表示不确定性越高，这里是否选取方向相反了呢？
​以上是代码复现时发现的疑问，如您能解惑将不胜感激！
